core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -2022,9 +2031,13 @@ private ExecutionModel lightlyValidateInsertModel(InsertModel model) throws SqlE
             throw SqlException.$(tableName.position, "literal expected");
         }
 
+        int rowTupleCount = model.getRowTupleCount();
         int columnSetSize = model.getColumnSet().size();
-        if (columnSetSize > 0 && columnSetSize != model.getColumnValues().size()) {
-            throw SqlException.$(model.getColumnPosition(0), "value count does not match column count");
+
+        for (int i = 0; i < rowTupleCount; i++) {
+            if (columnSetSize > 0 && columnSetSize != model.getRowValues(i).size()) {
+                throw SqlException.$(model.getColumnPosition(0), "row value count does not match column count [tuple=" + (i + 1) + "]");
```

Comment:
error position is incorrect - should position on tuple with incorrect value count